setup.sh

```diff
@@ -2,15 +2,23 @@
 set -e
 
 echo -n "Deploying git hooks: "
+
 HOOKDIR=.git/hooks
 
+# when repo is a subrepo, .git is a file describing the path to the actual gitdir
+if [ -f .git ]
+then
+    # reroute $HOOKDIR to the actual directory
+    HOOKDIR=$(perl -nle 'print $1 if /gitdir: (.+)/' .git)
```

Comment:
Is the `.git` file something like a symbolic link? I am worried about compatibility issues. Is `perl` installed on BBBs? Can we used something more standard, e.g. `readlink`?

Reply:
You're right. I completely overlooked the purpose of the git hooks. Since we won't be committing from the subrepo, there really is no need for the hooks.

The `.git` file is not a symlink but instead a text file with content
`
gitdir: ../.git/modules/hyped-pod_code/
`.

I used `perl` instead of `grep` to extract the file path because there are some inconsistencies with it on macOS causing it not to work properly.

Will update the file to ignore the hooks if the repo is a submodule.